.github/actions/commit_and_push/commit_and_push.ts

```diff
@@ -0,0 +1,187 @@
+import { execSync } from 'child_process';
+import { readFile, writeFile } from 'fs/promises';
+import { tmpdir } from 'os';
+import { join } from 'path';
+
+type CreateCommitInput = {
+    branch: {
+        repositoryNameWithOwner: string;
+        branchName: string;
+    };
+    fileChanges: {
+        additions: Addition[];
+    };
+    message: {
+        headline: string;
+    };
+    expectedHeadOid: string;
+};
+
+type Addition = {
+    path: string;
+    contents: string;
+};
+
+async function commitAndPush(): Promise<void> {
+    verifyEnvironmentVariables();
+
+    execSync(`git add ${process.env.ADD_FILES || '--all'}`, {
+        stdio: 'inherit'
+    });
+
+    const changedFiles = getChangedFiles();
+
+    if (changedFiles.length === 0) {
+        console.info('ℹ️ No changes to commit');
+        return;
+    }
+    console.info(`📝 Found ${changedFiles.length} changed files:`);
+    changedFiles.forEach((file) => console.info(`  - ${file}`));
+
+    const expectedHeadOid = getExpectedHeadOid();
+
+    const additions = await encodeFileContents(changedFiles);
+
+    const input = createInput(additions, expectedHeadOid);
+
+    await createCommitWithRetry(input);
+
+    console.info('✅ Commit created successfully!');
+}
+
+function verifyEnvironmentVariables(): void {
```

Comment:
Rename to something like `extractEnvironmentVariables`. Return new variables with the defined values.

Reply:
done